corehq/apps/couch_sql_migration/rebuildcase.py

```diff
@@ -18,17 +18,19 @@ def is_action_order_equal(sql_case, couch_json):
     """
     def dedup(items):
         return list(dict.fromkeys(items))
+    if was_rebuilt(sql_case):
+        return False
     sql_ids = dedup(t.form_id for t in sql_case.transactions if t.form_id)
     couch_ids = dedup(a["xform_id"] for a in couch_json["actions"] if a["xform_id"])
-    return sql_ids == couch_ids
+    return len(sql_ids) == len(couch_ids) and sql_ids != couch_ids
```

Comment:
I'm a little confused how it went from `sql_ids == couch_ids` to `<something> and sql_ids != couch_ids` without nearly reversing the meaning, maybe you could just fill in what I'm not seeing?

Reply:
Is this where you implement the item
- do not do SQL rebuild if transactions differ between couch and SQL
we talked about earlier?

I think this line is supposed to check that it's the same ids but just in a different order, which isn't strictly what it does. If I'm getting the intent right, maybe it's worth changing this to

```
    return set(sql_ids) == set(couch_ids) and sql_ids != couch_ids
```
?

Reply:
Good catch. I'll change that.